src/wp-includes/class-wp-textdomain-registry.php

```diff
@@ -105,81 +105,84 @@ public function set( $domain, $locale, $path ) {
 	}
 
 	/**
-	 * Resets the registry state.
+	 * Set the default path to the plugin's/theme's languages directory.
 	 *
-	 * @since 6.1.0
+	 * Used by {@see load_plugin_textdomain()} and {@see load_theme_textdomain()}.
+	 *
+	 * @param string $domain Text domain.
+	 * @param string $path   Language directory path.
 	 */
-	public function reset() {
-		$this->cached_mo_files = null;
-		$this->all             = array();
-		$this->current         = array();
+	public function set_default_path( $domain, $path ) {
+		$this->default_path[ $domain ] = untrailingslashit( $path );
 	}
 
 	/**
-	 * Gets the path to a translation file in the languages directory for the current locale.
+	 * Gets the path to the language directory for the current locale.
+	 *
+	 * Checks the plugins and themes language directories as well as any
+	 * custom directory set via {@see load_plugin_textdomain()} or {@see load_theme_textdomain()}.
 	 *
 	 * @since 6.1.0
 	 *
+	 * @see _get_path_to_translation_from_lang_dir()
+	 *
 	 * @param string $domain Text domain.
 	 * @param string $locale Locale.
-	 * @return string|false MO file path or false if there is none available.
+	 * @return string|false Language directory path or false if there is none available.
 	 */
 	private function get_path_from_lang_dir( $domain, $locale ) {
-		if ( null === $this->cached_mo_files ) {
-			$this->set_cached_mo_files();
+		$locations = array(
+			WP_LANG_DIR . '/plugins',
+			WP_LANG_DIR . '/themes',
+		);
+
+		if ( isset( $this->default_path[ $domain ] ) ) {
+			$locations[] = $this->default_path[ $domain ];
 		}
 
 		$mofile = "{$domain}-{$locale}.mo";
 
-		$path = WP_LANG_DIR . '/plugins/' . $mofile;
+		foreach ( $locations as $location ) {
+			if ( ! isset( $this->cached_mo_files[ $location ] ) ) {
+				$this->set_cached_mo_files( $location );
+			}
 
-		if ( in_array( $path, $this->cached_mo_files, true ) ) {
-			$path = WP_LANG_DIR . '/plugins/';
-			$this->set( $domain, $locale, $path );
+			$path = $location . '/' . $mofile;
 
-			return $path;
+			if ( in_array( $path, $this->cached_mo_files[ $location ], true ) ) {
+				$this->set( $domain, $locale, $location );
+
+				return trailingslashit( $location );
+			}
 		}
 
-		$path = WP_LANG_DIR . '/themes/' . $mofile;
-		if ( in_array( $path, $this->cached_mo_files, true ) ) {
-			$path = WP_LANG_DIR . '/themes/';
+		// If no path is found for the given locale anda default path has been set
+		// using load_plugin_textdomain/load_theme_textdomain, use that one.
+		if ( 'en_US' !== $locale && isset( $this->default_path[ $domain ] ) ) {
```

Comment:
I know there is a comment here, but can you explain why we need to hardcode this value?

Reply:
You mean why hardcoding the string `en_US`?

(Aside: this code is not new, just moved and slightly adapted).